### PR TITLE
I5150 hiding/404ing language tools link on android

### LINKED ISSUE #5150: Hide "Dictionaries and Language Packs" page on Android
In the future Firefox for Android might support some sort of language packs, but at the moment it doesn't. As pointed out in https://bugzilla.mozilla.org/show_bug.cgi?id=1462689 at best they do nothing, at worst they break about:addons. In https://github.com/mozilla/addons-server/issues/8381 we are going to make langpacks incompatible with Firefox for Android, so ultimately they won't show up in the "Dictionaries and Language Packs" page...

... But dictionaries are actually not that much more useful in Android. As far as I can tell installing one doesn't break anything, but doesn't do anything either. The only place where there is spellchecking in Firefox for Android seems to be the keyboard, which is not controlled by Firefox.

So, unless I'm wrong, I think the link to "Dictionaries and Language Packs" page should not be displayed for Android, and we should add a warning in that page if accessed through Android (in both cases, instead of doing UA sniffing inside the page, just checking that we're in a /android/ URL should be enough)

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #5150

hey @willdurand / @bobsilverberg , I actually haven't got an android to officially test this (yet - I will look into this) but when I use android in the url, the link doesn't appear. 

Please let me know if you think this is sufficient or if this needs to be tested on an actual android. In the meantime, I will look into an emulator/set up for this..

### CODE REVIEW

**Comment by rebmullin** [comment]
@willdurand @kumar303 @bobsilverberg  - can someone take a look?  This issue is a  P2. I am out on Weds for a week 🏖  so want to make sure this gets in ideally or at least your aware of this issue.  Please take a look when you can : )

File: tests/unit/amo/components/TestSectionLinks.js [hunk 1]
```diff
@@ -48,14 +48,26 @@ describe(__filename, () => {
...
+    const root = render({ viewContext: ADDON_TYPE_EXTENSION });
+
+    expect(
+      root.find('.SectionLinks-dropdownlink').findWhere((element) => {
+        return element.prop('to') === '/language-tools/';
+      }),
+    ).toHaveLength(0);
+  });
+
+  it('renders a link to the Language Tools page for non Android', () => {
```
Comment (willdurand) [inline tests/unit/amo/components/TestSectionLinks.js]: > for non-Android clients

File: tests/unit/amo/components/TestSectionLinks.js [hunk 2]
```diff
@@ -48,14 +48,26 @@ describe(__filename, () => {
...
+    _store.dispatch(setClientApp(CLIENT_APP_FIREFOX));
     const root = render({ viewContext: ADDON_TYPE_EXTENSION });
 
     expect(
       root.find('.SectionLinks-dropdownlink').findWhere((element) => {
         return element.prop('to') === '/language-tools/';
       }),
-    ).toHaveProp('children', 'Dictionaries & Language Packs');
```
Comment (willdurand) [inline tests/unit/amo/components/TestSectionLinks.js]: I guess we should keep this assertion.

File: src/amo/routes.js [hunk 3]
```diff
@@ -61,6 +61,7 @@ export default (
       )}
       <Route path=":visibleAddonType/categories/" component={CategoriesPage} />
       <Route path=":visibleAddonType/:slug/" component={Category} />
+      <Route path="/:lang/android/language-tools/" component={NotFound} />
```
[the remark this answers is not in the document]
Comment (rebmullin) [reply src/amo/routes.js]: @willdurand, sure thing...thx

just curious tho, how were you thinking this should be done? through ops? or?

[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/routes.js]: I would have returned a `NotFound` in the `LanguageTools` component.

[the remark this answers is not in the document]
Comment (rebmullin) [reply src/amo/routes.js]: ahh ok..

**Pull request merged**